R/data-combined.R

```diff
@@ -0,0 +1,867 @@
+#'   When providing the paths, only absolute full paths are supported (i.e., no
+#'   matching with '*' possible). If `quantitiesOrPaths` is `NULL` (default
+#'   value), returns the results for all output defined in the results.
+#' @param individualIds Numeric IDs of individuals for which the results
+#'   should be extracted. By default, all individuals from the results are
+#'   considered. If the individual with the provided ID is not found, the ID is
+#'   ignored.
+#' @param population Population used to calculate the `simulationResults`
+#'   (optional). This is used only to add the population covariates to the
+#'   resulting dataframe.
+#' @param dataSets Instance (or a `list` of instances) of the `DataSet`
+#'   class.
+#' @param groups A string or a list of strings assigning the data set to a
+#'   group. If an entry within the list is `NULL`, the corresponding data set is
+#'   not assigned to any group (and the corresponding entry in the dataframe
+#'   will be an `NA`). If provided, `groups` must have the same length as
+#'   `dataSets` and/or `simulationResults`. If no grouping is specified for any
+#'   of the dataset, the column `group` in the dataframe output will be all
+#'   `NA`.
+#' @param names A string or a list of string. This argument will be encountered
+#'   across different methods:
+#'
+#' - `$setDataTransformations()`: In the context of this method, a list of names
+#' specifying which observed datasets and/or paths in simulated dataset to
+#' transform with the specified transformations. Default is `NULL`, i.e., the
+#' transformations, if any specified, will be applied to all rows of the
+#' dataframe.
+#'
+#' - `$addSimulationResults()`: In the context of this method, a list of strings
+#' assigning new names to the quantities or paths present in the entered
+#' `SimulationResults` object. Note that the datasets whose names you wish to
+#' not change should be specified as `NULL` in the list.
+#'
+#' - `$addDataSets()`: In the context of this method, a list of strings
+#' assigning new names to the list of instances of the `DataSet` class. Note
+#' that the datasets whose names you wish to not change should be specified as
+#' `NULL` in the list.
+#'
+#'
+#' @import tidyr
+#' @import ospsuite.utils
+#'
+#' @examples
+#'
+#' # load the simulation
+#' simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simFilePath)
+#' simulationResults <- runSimulations(simulations = sim)
+#'
+#' # create a new dataset object
+#' dataSet <- DataSet$new()
+#'
+#' # created object with datasets combined
+#' myCombDat <- DataCombined$new()
+#' myCombDat$addSimulationResults(simulationResults)
+#' myCombDat$addDataSets(dataSet)
+#'
+#' # print the object
+#' myCombDat
+#' @docType class
+#' @export
+
+DataCombined <- R6::R6Class(
+  classname = "DataCombined",
+  inherit = Printable,
+
+  # public fields and methods ------------------------------------
```

Comment:
See my comment above. Once again, this has nothing to do with readability and everything to do with convenience of working with files with a large amount of code that needs to be quickly traversed.

Reply:
I understand but I think this could hide a design problem. There is absolutely no reason for any class to have 800 lines of code unless it is machine generated code.
The fact that this is required to be able to navigate the codes can (does not have to) mean that this class does too much. Of course, there is a  lot of validation stuff that does not belong there. Once this is removed, how big is the file. Do you still need this region?

Reply:
800 LOC involves hundreds of lines of comments, empty lines, functions that might be moved to the utilities package.
So the actual code base is much smaller than 800 LOC.

> Do you still need this region?

Before I answer this, I am actually curious as to why you are so opposed to having comments that act as flags in an IDE and give the file a navigation structure. How is that detrimental to anything?

Reply:
I am not opposed to it. I am traumatized with code using region maybe
https://github.com/Open-Systems-Pharmacology/PK-Sim/blob/develop/src/PKSim.UI/Views/ProteinExpression/ExpressionDataView.cs#L115
 
900 lines of code, lots of region, lots of comments, no one knows how to fix this because the code does a million unrelated tasks.

So my first instinct, as an old fashioned developer granted, is to frown upon code where regions are needed in order to navigate the code. It is not because the IDE supports it that is is a *good* thing. 

I do not say that this is bad or good. I am just saying that that relying on region to be able to navigate in the code *could* mean that the code does too much. <= Do you agree with this statement?

Reply:
and here some of the reasons why we do not use region in our code
https://softwareengineering.stackexchange.com/questions/53086/are-regions-an-antipattern-or-code-smell

I do not know the R world, and maybe this is different and much welcomed. My goal is to start a discussion to make sure we have at least talked about those different way of organizing code (As you said, this is purely about code organization)

Reply:
I would say, having the region markers for "public", "active", and "private" are ok because it is sometimes indeed not easy to navigate between them.

"setter methods", "getter methods", "print method" (sure it is a print method, if it is called `print`), "private methods" etc just add lines of code and introduce "too much structure" (or too many hierarchy levels), so I would get rid of them.